app/src/main/java/de/tum/in/tumcampusapp/component/ui/studyroom/StudyRoomGroupManager.kt

```diff
@@ -28,7 +28,15 @@ class StudyRoomGroupManager(context: Context) {
             groupsDao.insert(*groups.toTypedArray())
 
             groups.forEach { group ->
-                roomsDao.insert(*group.rooms.toTypedArray())
+                group.rooms.forEach { room ->
+                    // only insert rooms that have data
+                    if (room.code != ""
```

Comment:
you could use the `isBlank` or `isEmpty` functions. While this might work, it is not technically correct, as you must compare `strings` with .equal to check for value equality. (Now you are checking reference equality, which works, cause Java has a central String pool)